### ISSUE TITLE
Extend “archive” pattern to DMs

### ISSUE THREAD

**Issue by MitchExpensify** [Expensify/App#13426]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

[Coming from this Slack conversation](https://expensify.slack.com/archives/C01GTK53T8Q/p1670426898386029?thread_ts=1670271674.527989&cid=C01GTK53T8Q)

### Problem
Active accounts in DMs with closed accounts can still send messages and money requests, despite the closed accountholder explicitly taking the action to close their account. When this happens the closed accountholder receives the unread messages email summary delivered via the `NotifyOfflineUsersAboutActivity` job, just like they would if they still had an active account.  This creates a confusing experience for 1) the person that actively closed their account that doesn't expect to continue to receive messages and money requests after taking that action and 2) the active accountholder who's unaware that the other participant doesn't intend to respond or pay them back. 

<img width="1511" alt="image" src="https://user-images.githubusercontent.com/16232057/206460868-08f4e6bc-54fa-4059-9c6b-5e66b63b3de7.png">
<img width="1511" alt="image" src="https://user-images.githubusercontent.com/16232057/206460915-b0dd5686-7baa-47de-95b5-ecffd6fc34ba.png">

### Solution

Extend our "archived" chat pattern used for rooms and workspace chats to DMs, such that the ability to send additional messages or request money within the DM is no longer possible after a participant has explicitly closed their account. 

1. The chat name will show as archived in the UI i.e `Tom Rhys Jones (archived)` 
2. The archivedReason text box that blocks further chat input explains it i.e `This chat is archived because **Tom Rhys Jones** has closed their account.`
3. If the closed accountholder reopens their account the chat is unarchived (similar to how we archive/unarchive a policyExpenseChat when a member is removed/re-added to a policy). 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by JmillsExpensify** [comment]
I'm still on the fence as to whether this is a WAQ initiative, or whether it's scope for N7. So for instance, building on the discussion in Slack, I absolutely agree that we should decline/cancel IOUs when someone closes their account. That is what's causing bugs. This however, is more of a polish item helping people understand whether a person is or isn't available (because their account is closed). Accordingly, I would default to putting this on HOLD for after WAQ. Curious for thoughts on this!

**Comment by trjExpensify** [comment]
Yeah, I thought we already agreed in the thread to put this part on hold. I'm updating the title to reflect that.

**Comment by JmillsExpensify** [comment]
Nice, thanks!

**Comment by trjExpensify** [comment]
No change, Melv.

**Comment by JmillsExpensify** [comment]
Removed the WAQ hold. I think this is a good one we should find a champion for. It's a loophole we want to close.

**Comment by trjExpensify** [comment]
Thanks, Maria! I don't think we're quite ready for engineering yet. This was a bit of a placeholder for further planning/thought and we should likely wait first for Cristi's PRs that are fixing the broken reports on closing an account that has outstandingIOUs first which is where this came up - so I can retest a few scenarios. 

If we're error free after we're all said and done there, I could equally see an argument for :do-nothing: in the meantime, revisiting this a bit further down the line with more usage - or another minimal approach could be to just provide a system message in the DMs upon taking the action, such that the active participant is at least made aware of that change in state, but isn't fully blocked from sending chats/requests thereafter if there's a need for it. After all, you can send requests to anyone with a valid email/phone number in the first instance, so there's parity in preserving that.

**Comment by trjExpensify** [comment]
@JmillsExpensify ^^ do you have any thoughts on this?

**Comment by JmillsExpensify** [comment]
Sorry, looking!

**Comment by JmillsExpensify** [comment]
Alright a couple of things.

> we should likely wait first for Cristi's PRs that are fixing the broken reports on closing an account that has outstandingIOUs first which is where this came up - so I can retest a few scenarios.

Where is that PR? For some reason I thought that was already done, but it's not linked from what I can see. 

> If we're error free after we're all said and done there, I could equally see an argument for :do-nothing: in the meantime, revisiting this a bit further down the line with more usage - or another minimal approach could be to just provide a system message in the DMs upon taking the action, such that the active participant is at least made aware of that change in state, but isn't fully blocked from sending chats/requests thereafter if there's a need for it. After all, you can send requests to anyone with a valid email/phone number in the first instance, so there's parity in preserving that.

I don't know, I think the archive pattern is a pretty helpful way to help the user understand that nothing is going to happen when they are trying to do something with that particular person's DM. Generally, I think if @MariaHCD would like to take this on, including the pre-design, I think that'd be great. I do think this is the kind of polish we should be aiming for now that WAQ is stable and we can pursue a bunch of polish/features in parallel.

**Comment by trjExpensify** [comment]
Yeah, that PR is done back in mid-Jan. Jason and I caught up earlier and we briefly talked about this one. Given the pattern exists to enable anyone to begin a conversation with anyone if they have a valid email/phone number, archiving the DMs would be out of step with that. 

We could add a system message to clarify that an account has closed, but that a venture away from this issue OP and its initial purpose - to the point we should start afresh on that. We're not short of priorities at the moment, and this doesn't seem pressing enough to pursue. With that, I'm going to close this out and we can pick that up another time with more usage, perhaps moderation features being in the mix at that time and so on.

**Comment by JmillsExpensify** [comment]
Nice, agree with that. Thank you!
